src/classes/DataFactory.cls

```diff
@@ -1,15 +1,42 @@
 public abstract class DataFactory {
 
-	private Schema.DescribeSObjectResult dsr;
-	private sObjectType objectType;
+	private static Set<SObjectField> fieldsToPatch = new Set<SObjectField>();
+	private static sObjectType objectType;
     private static SObject defaultInstance;
+    private Integer countInstances = 0;
     
-    public DataFactory(SObjectType objectType) {
-        this.objectType = objectType;
-        this.dsr = objectType.getDescribe();
+    public DataFactory(SObjectType anObjectType) {
+        objectType = anObjectType;        
+        init();
+    }
+    
+    private void init() {
+    	Schema.DescribeSObjectResult dsr = objectType.getDescribe();
+    	Map<String, Schema.SObjectField> fieldMap = dsr.fields.getMap();
+        for (String fieldName : fieldMap.keySet()){
+   			SObjectField sField = fieldMap.get(fieldName);
+            Schema.DescribeFieldResult dField = sField.getDescribe();
+            Schema.DisplayType fldType = dField.getType();            
+            if (Schema.DisplayType.STRING.equals(fldType) || Schema.DisplayType.TextArea.equals(fldType)) {
+            	fieldsToPatch.add(sField);
+            }
+        }
+    }
+    
+    private void prePopulate(SObject item) {    	
+        Integer i = 0;
+        for (Schema.SObjectField sField : fieldsToPatch) {
+        	String value = (String) item.get(sField);
+        	if (value!=null) {
+        		List<String> fillers = new List<String> { String.valueOf(countInstances + i) };
+				String formatted = String.format(value, fillers);
+				item.put(sField, formatted);
+        	}
+		}
+		i++;
     }
```

Comment:
This variable doesn't seem to have a very meaningful name. I suggest to give this variable a concise but meaningful name that suggest its purpose.